Public/Src/Tools/Execution.Analyzer/Analyzers.Partner/PipExecutionPerformanceAnalyzer.cs

```diff
@@ -56,23 +63,67 @@ internal sealed class PipExecutionPerformanceAnalyzer : Analyzer
     {
         private const string Indentation = "    ";
         private readonly StreamWriter m_writer;
+        private readonly bool m_isCsvFormat;
+
         private readonly Dictionary<PipId, ProcessPipExecutionPerformance> m_processPerformance =
             new Dictionary<PipId, ProcessPipExecutionPerformance>();
 
+        private Dictionary<PipId, TimeSpan> m_stepDurations = new Dictionary<PipId, TimeSpan>();
+
         private string m_indent = string.Empty;
 
         /// <nodoc />
         [SuppressMessage("Microsoft.Reliability", "CA2000:Dispose objects before losing scope")]
-        public PipExecutionPerformanceAnalyzer(AnalysisInput input, string outputFilePath)
+        public PipExecutionPerformanceAnalyzer(AnalysisInput input, string outputFilePath, bool isCsvFormat)
             : base(input)
         {
             Contract.Requires(!string.IsNullOrWhiteSpace(outputFilePath));
 
             m_writer = new StreamWriter(outputFilePath);
+            m_isCsvFormat = isCsvFormat;
         }
 
-        /// <inheritdoc />
         public override int Analyze()
+        {
+            if (m_isCsvFormat)
+            {
+                ToCsv();

```

Comment:
nit: maybe rename the methods (this and ToJson) to reflect the kind of data we are printing?